internal/backends/winit/renderer/sw.rs

```diff
@@ -184,4 +192,29 @@ impl super::WinitCompatibleRenderer for WinitSoftwareRenderer {
         // and the buffer age doesn't respect that, so clean the partial rendering cache
         self.renderer.set_repaint_buffer_type(RepaintBufferType::NewBuffer);
     }
+
+    fn grab_window(
+        &self,
+        window: &i_slint_core::api::Window,
+    ) -> Result<SharedImageBuffer, PlatformError> {
+        let size = window.size();
+
+        let Some((width, height)) = size.width.try_into().ok().zip(size.height.try_into().ok())
+        else {
+            // Nothing to render
+            return Err("grab_window() called on window with invalid size".into());
+        };
+
+        // We could use Surface::fetch() here, but it's only implemented on X11 and Web.
+
+        let mut target_buffer =
+            SharedPixelBuffer::<i_slint_core::graphics::Rgb8Pixel>::new(width, height);
+
+        self.force_next_frame_new_buffer.set(true);
+        self.renderer.set_repaint_buffer_type(RepaintBufferType::NewBuffer);
+
+        let _ = self.renderer.render(target_buffer.make_mut_slice(), width as usize);
```

Comment:
you ignore the error instead of propagating it?

Reply:
render() doesn't return an error, but the region that's not needed here. That said, I see that I can move all this code away - found a trick :)